micro: copy operator FILL kernel from lite

rkuester opened the pull request:
This is a copy without modification of the kernel and test for
operator FILL from tensorflow/lite/kernels at 67993f46cb9.
Adaptations to micro and addition to the micro build to follow.

This is part of the work to port operator FILL as tracked in #45306.

rkuester committed:
micro: copy operator FILL kernel from lite

This is a copy without modification of the kernel and test for
operator FILL from tensorflow/lite/kernels at 67993f46cb9.
Adaptations to micro and addition to the micro build to follow.

petewarden commented on tensorflow/lite/micro/kernels/fill.cc:
```diff
@@ -0,0 +1,172 @@
+    case kTfLiteInt64:
+      return ResizeOutputImpl<int64_t>(context, dims, output);
+    default:
+      context->ReportError(
```
Instead of context->ReportError, can you use the TF_LITE_KERNEL_LOG macro instead, here and throughout the file? This allows us to strip the error strings when we need to minimize the binary size, and we have an internal check that blocks bare use of ReportError. You can see an example here:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/micro/kernels/activations.cc#L188

rkuester replied:
Will do. Is it okay if I do this in the next PR, since this PR is just the copy-without-modification from lite?

petewarden replied:
Unfortunately the internal checks fail with this line present, so we'll need to make this change in this PR (even though we were hoping it could be a straight copy).

rkuester replied:
Okay, I've attempted to fix this in c4d0fd2. Please let me know if it fails the internal check again, since I cannot see the check here on the outside.

advaitjain replied:
yeah, this should do the trick.

PR https://github.com/tensorflow/tensorflow/pull/46298 will make the error visible externally.

rkuester committed:
micro: fix formatting in inactive copy of op FILL

fill_test.cc as coped from lite does not pass a formatting check
during CI. Even though this file is not a part of the build, it
must still pass a formatting check to be merged. In this case,
use clang-format to fix the order of the inclusions.

rkuester committed:
micro: replace ReportError() with TF_LITE_KERNEL_LOG macro

Replace naked calls to ReportError() with the TF_LITE_KERNEL_LOG
macro to satisfy an internal CI check, even though this file is
not presently included in the build.

rkuester committed:
micro: delete gmock/gtest includes in copy of FILL

Delete the gmock and gtest includes in the currently inactive
copy of op FILL. There is an annoying difference between internal
and external versions of the code that causes clang-formatting
errors in CI checks (see issue #46297). When the code is imported
internally, the gmock and gtest includes become:

    #include "/internal/path/to/gmock/gmock.h"

and then clang-format wants them to be in the lower include block
(i.e., next to "flatbuffers/flatbuffers.h").

The solution for now is to delete gmock and gtest includes when
making these operator copies.
